### ISSUE TITLE
Grid info tracking

### ISSUE THREAD

**Issue by dschwoerer** [boutproject/BOUT-dev#2544]
We do currently do some tracking of grid infos [[1]](../tree/next/src/mesh/impls/bout/boutmesh.cxx)

I would like to do the same for zoidberg - should I just duplicate the code, or should we generalise the code?
e.g. `hypnotoad_version` -> `grid_software_version`?
@johnomotani any thoughts?

**Comment by johnomotani** [comment]
My vote would be just duplicate the code to add whatever things you want for zoidberg. I don't have strong feelings though - it feels a bit wrong not to separate the grid generator metadata stuff out into its own method or something, but in practice we're unlikely to have more than hypnotoad and zoidberg to look after so on balance just adding some lines for zoidberg seems like the simplest option to me.

**Comment by ZedThree** [comment]
+1 for duplicate for now, we could always refactor later. Ideally, one might imagine we had something like a list of `input_software` metadata with `software_name`, `software_version`, etc., so that we could record all the useful info for all the software that generated inputs.

I would say we should probably wrap this pattern up into a helper function:

```cpp
  if (this->get(grid_id, "grid_id") == 0 and not grid_id.empty()) {
    file.add(grid_id, "grid_id", false);
  }
```

**Comment by dschwoerer** [comment]
Thinking about it again - would it not be best to include the whole grid file? For FCI the grid file is anyway needed for plotting. However, I am not sure what the best approach would be for this ...

We could copy the grid file as `${grid_id}.nc` to the output folder?
Then it would be sufficient to copy the `grid_id` and when `xemc3.open_boutdataset()` could collect the grid.

Btw, the default values should probably be changed:
```
	Option grid_id =  (grids/rotating-ellipse-36x8x32-ix20.fci.nc) (default)
	Option hypnotoad_version =  (grids/rotating-ellipse-36x8x32-ix20.fci.nc) (default)
	Option hypnotoad_git_hash =  (grids/rotating-ellipse-36x8x32-ix20.fci.nc) (default)
	Option hypnotoad_git_diff =  (grids/rotating-ellipse-36x8x32-ix20.fci.nc) (default)
	Option hypnotoad_geqdsk_filename =  (grids/rotating-ellipse-36x8x32-ix20.fci.nc)
```
(I think this is an "old" grid without `grid_id`)

**Comment by johnomotani** [comment]
@dschwoerer I agree the grid file needs keeping. At the moment it is up to the user to keep track of this - `grid_id` was just intended to be able to check that you do have the right grid file. I'm not sure always copying the grid file to the output file is a good idea - for our workflow this would lead to hundreds of copies of a few grid file floating around. If it suits your workflow, I guess it shouldn't be too hard to make a copy of the grid file from your `PhysicsModel`?

In the long term, including either the grid file itself (as a NetCDF group?), or all the info from the grid file, in the dump files seems like a good idea. When I've thought about it it seems like a pain to implement at the moment: the current `DataFile` interface is not great for this (e.g. needing to pass persistent Field3D variables to be saved, although I guess it might be possible to work around this...); and `xarray` doesn't handle NetCDF groups nicely (yet!). So seems to me like something to bear in mind when refactoring the file I/O rather than something to worry about now.
